### ISSUE TITLE
Is there an Option for 'leaving/quit joshuto on current directory'?

### ISSUE THREAD

**Issue by theCollectiv** [kamiyaa/joshuto#230]
I really like joshuto and appreciate the work done by this project.

The default behavior - if you leave/quit joshuto - that you are in the directory, where you started the joshuto session.
[This behavior is the default behavior if you start the session with `ranger`]

Is there an option to set, that you can leave/quit joshuto on the current directory, that you are in?
[In ranger you can force this behavior by starting ranger with `. ranger`]
I really like the behavior and find it much more convenient than the default one ( 1. you maybe don't remember after hours, where you in which directory you did start the joshuto-session 2. If you are in a session, you can quickly go to terminal (and do some stuff there) and go back to joshuto.

**Comment by kamiyaa** [comment]
Yep, there is a bash script you can find here: https://github.com/kamiyaa/joshuto/blob/main/docs/configuration/keymap.toml.md#general

that allows you to exit to current directory

**Comment by Sonico98** [comment]
> Yep, there is a bash script you can find here: https://github.com/kamiyaa/joshuto/blob/main/docs/configuration/keymap.toml.md#general
> 
> that allows you to exit to current directory

Thanks! It works fine except if we use close_tab and there's a single tab present.

**Comment by thepenguinthatwants** [comment]
Can you give example of how to use this script?
I added it to my bashrc, but opening joshuto and navigating to another folder doesn't seem to help. When I quite joshuto it just goes back to the original folder where I opened it.

**Comment by thepenguinthatwants** [comment]
Ah seems like quitting with Q should work as it created temp folder, but instead I am getting Exit code: 101, is there a way to make this more verbose?

**Comment by kamiyaa** [comment]
I've add more documentation on how `quit` works  and the exit codes of each one here: https://github.com/kamiyaa/joshuto/blob/main/docs/configuration/keymap.toml.md#quit-quit-joshuto

lmk if it helps!

**Comment by JohannesProgrammiert** [comment]
May this helps others like me who come from ranger and don't like reading docs.

To make a fresh installation of joshuto always exit to the current directory, do this:

1. Make an alias in your .bashrc or .zshrc:
    
    ```bash
    alias joshuto='joshuto --output-file /tmp/joshutodir; LASTDIR=`cat /tmp/joshutodir`; cd "$LASTDIR"'
    ```
2. Make sure you copied the `.toml` files from `config` (in this repo) to `$HOME/.config/joshuto`.
3. Edit `$HOME/.config/joshuto/keymap.toml` and find

    ```
    { keys = [ "q" ],		command = "close_tab" },
    { keys = [ "Q" ],		command = "quit --output-current-directory" },
    ```

    and change it to

    ```
    { keys = [ "Q" ],		command = "close_tab" },
    { keys = [ "q" ],		command = "quit --output-current-directory" },
    ```

4. (optional) Initially, I get a a lot of warnings from joshuto, so I removed every line with `escape`, `numbered_command`, `symlink_files`, `toggle_visual`, `linemode` from `$HOME/.config/joshuto/keymap.toml`.

Then joshuto will exit to the current directory by pressing `q`.

Probably there is better way but as a user I am just looking for snippet I can copy to my config.

**Comment by sorathpanzer** [comment]
My friends for those who use nushell, and want to exit to the current directory, I left here the function to add to ~/.config/nushell/config.nu:

```
def --env joshuto [...args] {
    let id = $nu.pid
    let tmp_dir = $"/tmp/($env.USER)"
    let output_file = $"($tmp_dir)/joshuto-cwd-($id)"

    if not ($tmp_dir | path exists) { mkdir $tmp_dir }

    try { ^joshuto --output-file $output_file ...$args } catch { }
    let exit_code = $env.LAST_EXIT_CODE

    match $exit_code {
        0 => {
            # regular exit, do nothing
        }
        101 => {
            if ($output_file | path exists) {
                let cwd = (open $output_file | str trim)
                cd $cwd
            }
        }
        102 => {
            # output selected files (not handled, same as bash)
        }
        _ => {
            print $"Exit code: ($exit_code)"
        }
    }
}
```
